nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -233,15 +254,24 @@ in {
       before = [ "ipfs.service" "ipfs-offline.service" "ipfs-norouting.service" ];
 
       preStart = ''
-        install -m 0755 -o ${cfg.user} -g ${cfg.group} -d ${cfg.dataDir}
+        install -m 0755 -o "${cfg.user}" -g "${cfg.group}" -d "${cfg.dataDir}"
       '' + optionalString cfg.autoMount ''
-        install -m 0755 -o ${cfg.user} -g ${cfg.group} -d ${cfg.ipfsMountDir}
-        install -m 0755 -o ${cfg.user} -g ${cfg.group} -d ${cfg.ipnsMountDir}
+        # XXX Sometimes, the daemon stops or crashes without unmounting.
+        umount -l "${cfg.ipfsMountDir}" || true
```

Comment:
Why a lazy unmount?

Reply:
If some process has an open handle in there, I don't want that to prevent a remount. This is a FUSE virtual FS, anyway, so I can't imagine how any harm could result from this.